config/crd/bases/ols.openshift.io_configs.yaml

```diff
@@ -0,0 +1,201 @@
+                            type: string
+                          maxMemory:
+                            anyOf:
+                            - type: integer
+                            - type: string
+                            description: Redis maxmemory
+                            x-kubernetes-int-or-string: true
+                          maxMemoryPolicy:
+                            default: allkeys-lru
+                            description: 'Redis maxmemory policy. Default: "allkeys-lru"'
+                            type: string
+                          port:
+                            default: 6379
+                            description: 'Redis port. Default: "6379"'
+                            maximum: 65535
+                            minimum: 1
+                            type: integer
+                        type: object
+                      type:
+                        default: memory
+                        description: 'Conversation cache type. Default: "memory"'
+                        enum:
+                        - redis
+                        - memory
+                        type: string
+                    type: object
+                  defaultModel:
+                    description: Default model for usage
+                    type: string
+                  defaultProvider:
+                    description: Default provider for usage
+                    type: string
+                  enableDebugUI:
+                    default: false
+                    description: 'Enable debug UI. Default: "false"'
+                    type: boolean
+                  logger:
+                    description: Logger settings
+                    properties:
+                      logFilename:
+                        description: Log filename
+                        type: string
+                      logLevel:
+                        default: INFO
+                        description: 'Log level. Default: "INFO"'
+                        type: string
+                      logSize:
+                        description: Log file size
+                        type: integer
+                    type: object
+                  summarizerModel:
+                    description: Summarizer model name
+                    type: string
+                  summarizerProvider:
+                    description: Summarizer provider name
+                    type: string
+                  validatorModel:
+                    description: Validator model name
+                    type: string
+                  validatorProvider:
+                    description: Validator provider name
+                    type: string
+                  yamlModel:
+                    description: YAML model name
+                    type: string
+                  yamlProvider:
+                    description: YAML provider name
+                    type: string
+                type: object
+              replicas:
```

Comment:
i think we should nest the fields that refer specifically to configuring the ols deployment object, under its own level.

e.g. ols.deployment.replicas, ols.deployment.resources, etc

as we are likely to continue to grow the set of deployment fields we allow to be configured from here, so having them all isolated within a hierachy level will make it more readable